src/components/filters/DateFilter/index.tsx

```diff
@@ -35,34 +37,81 @@ const FieldWrapper = styled.div`
 const Date = styled(Textfield)`
   font-size: 14px;
   height: 32px;
+  padding-left: 6px;
+  padding-right: 6px;
 `
 
 interface Props {
-  onChangeFrom: () => void
-  onChangeTo: () => void
-  onSubmit: () => void
+  onChangeFrom?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onChangeTo?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onSubmit: (from: number, to: number) => void
   title: string
 }
 
 export const DateFilter: React.FC<Props> = (props) => {
   const { onChangeFrom, onChangeTo, onSubmit, title } = props
 
+  const [from, setFrom] = React.useState<Maybe<number>>(null)
+  const [to, setTo] = React.useState<Maybe<number>>(null)
+
+  const onChangeFromInternal = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (typeof onChangeFrom === 'function') {
+      onChangeFrom(event)
+    }
+
+    const fromDateMoment = moment(event.currentTarget.value)
+    const fromTimestamp = fromDateMoment.unix()
+
+    setFrom(fromTimestamp)
+  }
+
+  const onChangeToInternal = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (typeof onChangeTo === 'function') {
+      onChangeTo(event)
+    }
+
+    const toDateMoment = moment(event.currentTarget.value)
+    const toTimestamp = toDateMoment.unix()
+
+    setTo(toTimestamp)
+  }
+
+  const onSubmitInternal = () => {
+    if (from && to) onSubmit(from, to)
+  }
+
+  // This clear the filters
+  React.useEffect(() => {
+    if (!from || !to) onSubmit(0, 0)
+  }, [from, to, onSubmit])
+
+  const errorMessage = React.useMemo(
+    () => (to && from && to < from ? 'To should be greater than From' : null),
+    [from, to]
+  )
+  const emptyValues = !from || !to
+
```

Comment:
(This also applies to min, max).
emptyValues can be wrapped in `useMemo` and use in the effect leveraging morgan's law. It would be `if(!emptyValues) ... ` .

Reply:
Done!